test/Microsoft.Azure.WebJobs.Extensions.Kafka.EndToEndTests/KafkaEndToEndTests.cs

```diff
@@ -265,14 +266,33 @@ public async Task Multiple_Hosts_Process_Events_At_Least_Once()
             var producerHost = await this.StartHostAsync(typeof(KafkaOutputFunctions));
             var producerJobHost = producerHost.GetJobHost();
 
-            var producerTask = producerJobHost.CallOutputTriggerStringAsync(
-                GetStaticMethod(typeof(KafkaOutputFunctions), nameof(KafkaOutputFunctions.SendToStringTopic)),
-                this.endToEndTestFixture.StringTopicWithTenPartitions.Name,
-                Enumerable.Range(1, producedMessagesCount).Select(x => EndToEndTestExtensions.CreateMessageValue(messagePrefix, x)), 
-                TimeSpan.FromMilliseconds(100));
+            var host2HasPartitionsSemaphore = new SemaphoreSlim(0);
 
-            IHost host1 = null, host2 = null;
+            // Split the call in 4, waiting 1sec between calls
+            var producerTask = Task.Run(async () => 
+            {
+                var allMessages = Enumerable.Range(1, producedMessagesCount).Select(x => EndToEndTestExtensions.CreateMessageValue(messagePrefix, x));
+                const int loopCount = 4;
+                var itemsPerLoop = producedMessagesCount / loopCount;
+                for (var i=0; i < loopCount; ++i)
+                {
+                    var messages = allMessages.Skip(i * itemsPerLoop).Take(itemsPerLoop);
+                    await producerJobHost.CallOutputTriggerStringAsync(
+                        GetStaticMethod(typeof(KafkaOutputFunctions), nameof(KafkaOutputFunctions.SendToStringTopic)),
+                        this.endToEndTestFixture.StringTopicWithTenPartitions.Name,
+                        messages);
+
+                    if (i == 0)
+                    {
+                        // wait until host2 has partitions assigned
+                        Assert.True(await host2HasPartitionsSemaphore.WaitAsync(TimeSpan.FromSeconds(30)));
```

Comment:
Could you add a failure message here so if/when the semaphore fails to release w/in 30 seconds we have a quick indicator in the test output that this was the cause of the failure?